Std/Data/List/Lemmas.lean

```diff
@@ -5,6 +5,7 @@ Authors: Parikshit Khanna, Jeremy Avigad, Leonardo de Moura, Floris van Doorn, M
 -/
 import Std.Control.ForInStep.Lemmas
 import Std.Data.List.Basic
+import Std.Data.Nat
```

Comment:
This is a cross cutting import, and also not a precise import. The needed theorem(s) should move to `Std.Data.Nat.Init.Lemmas` and that should be imported here.

Reply:
Is it possible to move the needed lemmas up into an `Init` folder, to reduce the imports needed here?